dashboard/app/controllers/api/v1/peer_review_submissions_controller.rb

```diff
@@ -60,7 +61,15 @@ def index
 
     # This query gets matching fully-hydrated models in the correct order.
     real_reviews = PeerReview.find(reviews.map(&:id))
+
+    # Pull out any that are tied to deprecated scripts/units
+    active_reviews = []
     real_reviews.each do |review|
+      script = Script.find(review.script_id)
+      active_reviews << review unless script.deprecated?
```

Comment:
Looks like change results in looping through lots of reviews twice, which if we're already loading all reviews probably isn't great...I think you could use the existing loop, and just add this at the top?

`next if review.script&.deprecated?`

Reply:
I see your point about iterating twice. I came up with a slightly different implementation, but it works and only loops once. Thanks for pointing this out!